### PR TITLE
build: fix VSCode problem matcher for the build task 

### PR DESCRIPTION
**Pull request by bajtos**
TypeScript compiler enables `pretty` mode when using the build mode (project references). Unfortunately, pretty mode is not supported by VSCode's problem matcher and therefore VSCode is not able to parse the errors to populate the Problems window.

This change addes `--pretty false` arguments to `tsc` when invoked from VSCode task "Build project".

In order to make this work, I had to also fix `lb-tsc` to correctly recognize and pass boolean arguments with a value. Before this change, `lb-tsc -b --pretty false` would discard `--pretty false` arguments.

## Checklist

👉 [Read and sign the CLA (Contributor License Agreement)](https://cla.strongloop.com/agreements/strongloop/loopback-next) 👈

- [ ] `npm test` passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [ ] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [ ] API Documentation in code was updated
- [ ] Documentation in [/docs/site](../tree/master/docs/site) was updated
- [ ] Affected artifact templates in `packages/cli` were updated
- [ ] Affected example projects in `examples/*` were updated

👉 [Check out how to submit a PR](https://loopback.io/doc/en/lb4/submitting_a_pr.html) 👈

### CODE REVIEW

File: packages/build/bin/compile-package.js [hunk 1]
```diff
@@ -164,8 +164,10 @@ function validArgsForBuild(args) {
   });
   let validArgs = args;
   if (args.includes('-b') || args.includes('--build')) {
-    validArgs = filterArgs(args, arg => {
-      if (validBooleanOptions.includes(arg)) return 1;
+    validArgs = filterArgs(args, (arg, next) => {
+      if (validBooleanOptions.includes(arg)) {
+        return next === 'false' || next === 'true' ? 2 : 1;
+      }
```
Comment (raymondfeng) [inline packages/build/bin/compile-package.js:170]: Does `tsc` allow `no-prettier`?

File: packages/build/bin/compile-package.js [hunk 2]
```diff
@@ -185,7 +187,7 @@ function filterArgs(args, filter) {
   const validArgs = [];
   let i = 0;
   while (i < args.length) {
-    const length = filter(args[i]);
+    const length = filter(args[i], args[i + 1]);
```
Comment (jannyHou) [inline packages/build/bin/compile-package.js:190]: wouldn't `args[i+1]` exceeds the array's index? 
e.g. `args.length = 5`, `i = 4`, `args[5]` doesn't exist

Comment (emonddr) [reply packages/build/bin/compile-package.js:190]: you need `args.length -1`

File: packages/build/bin/compile-package.js (hunk 1 above)
Comment (bajtos) [reply packages/build/bin/compile-package.js:170]: No, it does not.

```
$ tsc -b --no-prettier
error TS5072: Unknown build option '--no-prettier'.
```

File: packages/build/bin/compile-package.js (hunk 2 above)
Comment (bajtos) [reply packages/build/bin/compile-package.js:190]: > wouldn't `args[i+1]` exceeds the array's index? 
> e.g. `args.length = 5`, `i = 4`, `args[5]` doesn't exist

Yes, `i+1` can exceed the array index. That's fine, `args[i+1]` will be resolved as `undefined` in such case.

```
$ node
> list = [1,2,3]; list[5]
undefined
```

**Comment by bajtos** [comment]
@raymondfeng @jannyHou @emonddr I responded to your comments, PTAL. Does the patch LGTY now?

**Pull request merged**